app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(params.require(:comment).permit(:text))
+    @comment.author = current_user
+    @comment.post_id = params[:post_id]
+
+    if @comment.save
+      flash[:success] = 'Comment added successfully!'
+      redirect_to request.referrer
+    else
+      flash.now[:error] = 'comment creation failed!'
+    end
+  end
+end
```

Comment:
[OPTIONAL]  In order to increase the security of data sent through the forms kindly consider using strong params to specify in the controller which parameters are accepted and used just as you have done in the posts controller. You can check [this link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md) for further explanation on how to use the params.